subprojects/base-services/src/main/java/org/gradle/internal/progress/BuildOperationDescriptor.java

```diff
@@ -21,21 +21,27 @@
 /**
  * Meta-data about a build operation.
  */
-public class BuildOperationDetails {
-    private final BuildOperationExecutor.Operation parent;
+public class BuildOperationDescriptor {
+    private final Long id;
```

Comment:
I felt we should move away from `Object id`. We could also use `org.gradle.internal.logging.events.OperationIdentifier` everywhere. But that basically just wraps a Long. And since the ID is passed around more and more (build scan plugin, worker processes) `Long` might be a simple and pragmatic solution here.

I see that there is also discussion about this in the context of #1838. We could also address it there or later (and change it back to `Object` for now).